### DISCUSSION TITLE
Bitwise operator in error handler

### DISCUSSION THREAD

**Discussion by mkluczka** [yiisoft/yii2#20251] [General]
I've been doing some debugging with Yii2, and found out my warning is silenced by this error handler

https://github.com/yiisoft/yii2/blob/master/framework/base/ErrorHandler.php#L248

```
 if (error_reporting() & $code) {
```

what is the purpose of bitwise operator used here? Or maybe it was supposed to be `&&` ?

**Comment by rhertogh** [comment]
Hi @mkluczka,
To me it looks like the bitwise and operator is correct.
The [error_reporting() function](https://www.php.net/manual/en/function.error-reporting.php) returns the current [error reporting level](https://www.php.net/manual/en/errorfunc.configuration.php#ini.error-reporting) which is a bit field. The `$code` parameter is the level of the error raised. So the `if` statement checks if the specific reporting level is enabled by doing the bitwise and operation.

If your warning is ignored by the `if` statement, you're current error reporting level probably doesn't include the `E_WARNING` level.
You can change it via the `php.ini` file by setting `error_reporting = E_ALL` or at runtime via by calling `error_reporting(E_ALL);`.
You can find the different error levels [here](https://www.php.net/manual/en/errorfunc.constants.php).
